### DISCUSSION TITLE
Wall shear stress BC in INSAD module

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#24958] [Q&A Modules: Navier-Stokes]
Hi all,

I am trying to understand how one might impose a wall shear stress BC in the finite element Navier-Stokes module - or even if it makes sense to do so.

Since we have the relation wall shear stress = viscosity * du/dy where du and dy are parallel and perpendicular to the wall respectively, it made sense to try this with a Neumann BC. 
I had a go at something similar to ADVectorFunctionNeumannBC, but instead of a function, passed in a vector containing wall shear components divided by viscosity.
However, the BC seemed to have no effect on the simulation, even if I changed the scale factors to be very large.

Since we might already have a no slip condition, I am having trouble understanding whether we can meaningfully impose a wall shea BC as well here - or in finite element at all. 
If we were to do this, is using an integrated BC in this manner the way to go? 

I should add that I am using the vector variable versions of the INSAD kernels, which I believe use the Laplace form of the INS equations.

Thanks!

**Comment by lindsayad** [comment]
So the weak term at the boundary for a given momentum component will be (for the Laplace form):
```
test * (n * mu * grad(u_comp) + n(comp) * pressure)
```
if you integrate the pressure by parts. If you don't integrate pressure by parts then you drop the second term. So if you want to supply a particular shear stress, then you can supply your vector of shear stresses and they should *not* be divided by the viscosity. And if you are integrating the pressure by parts, then you will also need to add a BC to handle that, e.g. a BC like:
```
test * n * pressure
```
for the vector INSAD implementation
